### PR TITLE
Normalize C:\\\\ to C:\ wrt #12

### PR DESCRIPTION
**Pull request by hasufell**
@Bodigrim

### CODE REVIEW

File: System/FilePath/Internal.hs [hunk 1]
```diff
@@ -984,9 +985,12 @@ normaliseDrive :: FILEPATH -> FILEPATH
 normaliseDrive bs
   | null bs = mempty
   | isPosix = pack [pathSeparator]
-  | otherwise = if isJust $ readDriveLetter x2
-         then map toUpper x2
-         else x2
+  | Just (drv, _) <- readDriveLetter x2
+                  = case unpack drv of
+                      (x:_:[]) -> pack [toUpper x, _colon]
+                      (x:_) -> pack [toUpper x, _colon, pathSeparator]
```
Comment (Bodigrim) [inline System/FilePath/Internal.hs:991]: Is it expected to have anything else than `:` as the second element of the list?

Comment (hasufell) [reply System/FilePath/Internal.hs:991]: No, but `readDriveLetter` is maintaining this invariant:

```
readDriveLetter :: STRING -> Maybe (FILEPATH, FILEPATH)
readDriveLetter bs = case unpack bs of
  (x:c:y:xs)
    | c == _colon && isLetter x && isPathSeparator y
    -> Just $ addSlash (pack [x,_colon]) (pack (y:xs))
  (x:c:xs)
    | c == _colon && isLetter x
    -> Just (pack [x,_colon], pack xs)
  _ -> Nothing
```

**Comment by hasufell** [comment]
It appears GHC 9.4.4 changed exception text leading to some property test failures.

File: System/FilePath/Internal.hs (hunk 1 above)
Comment (hasufell) [reply System/FilePath/Internal.hs:991]: We could inline this, but...

**Review by Bodigrim — APPROVED** [review]
Works for me!

**Pull request merged**